app/models/post.rb

```diff
@@ -0,0 +1,15 @@
+class Post < ApplicationRecord
+  belongs_to :author, foreign_key: :author_id, class_name: 'User'
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+
+  after_save :update_user_post_counter
+
+  def update_user_post_counter
+    author.increment!(:posts_counter)
+  end
```

Comment:
[OPTIONAL] - Good job using one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to database 👍 

- [ ] Also, I would suggest that we make the `update_post_counter` method private

```ruby
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```

- [ ] It is considered good practice to declare callback methods as private. If left public, they can be called from outside of the model and violate the principle of object encapsulation 🤗

Reply:
Thank you @abdulhamiid  for the feedback.